### ISSUE TITLE
completion callback called too many times

### ISSUE THREAD

**Issue by amcdnl** [dewe/nginx-server#2]
When I start it from gulp i get the following:

```
[09:25:10] 'start:web' errored after 719 ms
[09:25:10] Error: task completion callback called too many times
  at finish (C:\dev\myapp\Web\node_modules\orchestrator\lib\runTask.js:15:10)
  at cb (C:\dev\myapp\Web\node_modules\orchestrator\lib\runTask.js:29:3)
  at C:\dev\myapp\Web\tasks\server.js:34:5
  at Socket.<anonymous> (C:\dev\myapp\Web\node_modules\nginx-server\lib\nginx.js:69:17)
  at emitOne (events.js:95:20)
  at Socket.emit (events.js:182:7)
  at readableAddChunk (_stream_readable.js:153:18)
  at Socket.Readable.push (_stream_readable.js:111:10)
  at Pipe.onread (net.js:529:20)
```

here is the setup:

```
gulp.task('start:web', function(done){

  var server = nginx({
    config: path.resolve(__dirname, '../', 'conf/nginx.conf')
  });

  var isDone = false;
  server.start(function () {
    console.log('started');
   done();
  });

  server.stop(function () {
    console.log('stopped');
  });
});
```

**Comment by dewe** [comment]
Long shot: you've got an nginx.conf with `worker_processes 2;` or even more?

I've been running this in a custom test setting only, with a single worker process. The [code triggering the callback](https://github.com/dewe/nginx-server/blob/master/lib/nginx.js#L67) is a bit naïve. :children_crossing: 

Can you try to change to a single worker process and see if it works? If so, then it's easy for me to fix that bug here.

**Comment by amcdnl** [comment]
I have mine set to `auto`, however, that should not affect the gulp callback I don't think?

**Comment by dewe** [comment]
Oh yes it will: the server.start callback will be called multiple times, each time calling gulp's done callback... and that makes gulp throw up.

**Comment by amcdnl** [comment]
@dewe what changed? its really hard to tell with the whitespace diff.

**Comment by amcdnl** [comment]
FYI - This did fix it! did not fix #3 though.

**Comment by dewe** [comment]
Oh, that's an ugly diff. :scream_cat: Great to hear it worked for you.

Basically, this is a one line change in server.start: https://github.com/dewe/nginx-server/blob/master/lib/nginx.js#L68 along with a new test: https://github.com/dewe/nginx-server/blob/master/test/nginx_server.js#L54.

Rest is mostly plumbing to have tests run in a Docker container.
